zebra-consensus/src/primitives/ed25519.rs

```diff
@@ -120,36 +119,41 @@ impl Verifier {
 
     /// Flush the batch using a thread pool, and return the result via the channel.
     /// This function returns a future that becomes ready when the batch is completed.
-    fn flush_spawning(batch: BatchVerifier, tx: Sender) -> impl Future<Output = ()> {
+    async fn flush_spawning(batch: BatchVerifier, tx: Sender) {
+        let (rsp_tx, rsp_rx) = tokio::sync::oneshot::channel();
+
         // Correctness: Do CPU-intensive work on a dedicated thread, to avoid blocking other futures.
-        tokio::task::spawn_blocking(|| {
+        rayon::spawn_fifo(|| {
```

Comment:
This changes panic behaviour:
https://docs.rs/rayon/latest/rayon/fn.scope_fifo.html#panics
https://docs.rs/rayon/latest/rayon/fn.spawn_fifo.html#panic-handling

Are we setting the rayon panic handler to our standard panic handler?

Reply:
> Are we setting the rayon panic handler to our standard panic handler?

I think we should, it could help with debugging later.

Reply:
I have a preference for panicking at the site of the problem, but it's not a blocker.

Also in this specific case, the code we're calling is low risk.

Reply:
> Are we setting the rayon panic handler to our standard panic handler?

Actually, it appears it is using the std panic hook.

Reply:
Setting the rayon panic hook seems like something for another ticket or PR.

Reply:
> Setting the rayon panic hook seems like something for another ticket or PR.

I meant that it seems to be printing ideal diagnostic information already:

![image](https://github.com/ZcashFoundation/zebra/assets/5491350/66ea4ad6-c4bc-4444-82fe-7d30a85fd50b)

Reply:
Thanks good, thanks for checking!